coordinator/src/block_streams/synchronise.rs

```
@@ -0,0 +1,599 @@
+use registry_types::StartBlock;
+
+use crate::indexer_config::IndexerConfig;
+use crate::redis::RedisClient;
+use crate::registry::IndexerRegistry;
+
+use super::handler::{BlockStreamsHandler, StreamInfo};
+
+pub async fn synchronise_block_streams(
+    indexer_registry: &IndexerRegistry,
+    redis_client: &RedisClient,
+    block_streams_handler: &BlockStreamsHandler,
+) -> anyhow::Result<()> {
+    let mut active_block_streams = block_streams_handler.list().await?;
+
+    for (account_id, indexers) in indexer_registry.iter() {
+        for (function_name, indexer_config) in indexers.iter() {
+            let active_block_stream = active_block_streams
+                .iter()
+                .position(|stream| {
+                    stream.account_id == account_id.to_string()
+                        && &stream.function_name == function_name
+                })
+                .map(|index| active_block_streams.swap_remove(index));
+
+            let _ = synchronise_block_stream(
+                active_block_stream,
+                indexer_config,
+                redis_client,
+                block_streams_handler,
+            )
+            .await
+            .map_err(|err| {
+                tracing::error!(
+                    account_id = account_id.as_str(),
+                    function_name,
+                    version = indexer_config.get_registry_version(),
+                    "failed to sync block stream: {err:?}"
```

Swallow errors, they will be retired on the next control loop